### PR TITLE
Make find operation argument more specific

### PR DESCRIPTION
**Pull request by zhangxin511**
## Description

> Make the find arguments more explicit

## Steps to Reproduce Bug and Validate Solution

> Our mongodb implemention has a implicit limit of 2000, which is quite confusing and not documented anywhere. 
> Added optional arguments to the contract, which gives consumer ability to change the runtime behavior and have a 
> knowledge in mind that there are limits.

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] My code follows the code style of this project.
-   [ ] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [ ] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

> If this introduces a breaking change, please describe the impact and migration path for existing applications below.

## Testing

> -   No breaking changes, should have performed like before. Validated using clicker

## Any relevant logs or outputs

> -   N/A

## Other information or known dependencies

> -   N/A

### CODE REVIEW

File: server/routerlicious/packages/services-core/src/database.ts [hunk 1]
```diff
@@ -51,9 +51,11 @@ export interface ICollection<T> {
      *
      * @param query - data we want to find
      * @param sort - object with property we use to sort on, whose value is 0 for descending order and 1 for ascending
+     * @param limit - optional, if set, query result will set less equal to this number. Our mongo layer internally used 2000.
```
Comment (hedasilv) [inline server/routerlicious/packages/services-core/src/database.ts:54]: nit: "optional - if set, limits the number of documents/records the cursor will return"

Borrowed and updated the description from here: https://www.mongodb.com/docs/drivers/node/v4.6/fundamentals/crud/read-operations/limit/?_ga=2.68335960.811589495.1653330422-1860421701.1644639209

File: server/routerlicious/packages/services-core/src/database.ts [hunk 2]
```diff
@@ -51,9 +51,11 @@ export interface ICollection<T> {
...
+     * @param skip - optional, if set, query result will skip the first few numbers based on number and sort values.
```
Comment (hedasilv) [inline server/routerlicious/packages/services-core/src/database.ts:55]: nit: "skip - optional. If set, defines the number of documents to skip in the results set"

Borrowed the description from here: https://www.mongodb.com/docs/manual/reference/method/cursor.skip/

**Review by hedasilv — APPROVED** [review]

File: server/routerlicious/packages/services-core/src/database.ts (hunk 1 above)
Comment (zhangxin511) [reply server/routerlicious/packages/services-core/src/database.ts]: Addressed. Kept the `2000` part.

File: server/routerlicious/packages/services-core/src/database.ts (hunk 2 above)
Comment (zhangxin511) [reply server/routerlicious/packages/services-core/src/database.ts]: Addressed.

**Review by hedasilv — APPROVED** [review]

**Pull request merged**